### PR TITLE
Add UR robot support

### PR DESCRIPTION
**Pull request by jk-ethz**
This PR adds support for velocity controlled UR robots, specifically the UR5e.
As it should run off the same code base and the UR has 6 joints compared to the 7 of Panda, many things are generalized in this PR. Most notably, the `moma_ocs2_ros` is now a templated header-only library.
The gains for the UR are tuned (lowered) according to the velocity control approach and allow a safe operation without noise.

- For review, it's easiest to first check the changes to [`definitions.h`](https://github.com/ethz-asl/moma/compare/projects/piloting...projects/piloting_feature/ur_support?expand=1#diff-ddd2a570a0729339ebcdb044b0e9195c4f1e8bc236ccfb7d3e20539ebde6a673), which introduce helper functions for getting state and input dimensions from the configurable arm input dimension. These are used throughout the code.
- There are some "TODO"s that are labeled as hard-coded. These are at places where we have no direct ability to get the joint count from anywhere else. So they are just kept like this for now.
- Since we omitted the frame transforms relying on the transform buffer due to controller crashes, [a proper checking of world frame is now implemented](https://github.com/ethz-asl/moma/blob/16b269406cb379b6df69a079c7beae6420d06f57/moma_controllers/moma_ocs2_ros/include/moma_ocs2_ros/mpc_velocity_controller.h#L356).

### CODE REVIEW

File: moma_controllers/moma_ocs2/include/moma_ocs2/MobileManipulatorPinocchioMapping.h [hunk 1]
```diff
@@ -39,19 +39,20 @@ template <typename SCALAR>
...
-  MobileManipulatorPinocchioMapping(BaseType baseType = BaseType::none) : baseType_(baseType) {};
+  MobileManipulatorPinocchioMapping(const size_t armInputDim, const BaseType baseType = BaseType::none) : armInputDim_(armInputDim), baseType_(baseType) {};
   ~MobileManipulatorPinocchioMapping() override = default;
   MobileManipulatorPinocchioMapping<SCALAR>* clone() const override { return new MobileManipulatorPinocchioMapping<SCALAR>(*this); }
 
   vector_t getPinocchioJointPosition(const vector_t& state) const override { return state; }
 
   vector_t getPinocchioJointVelocity(const vector_t& state, const vector_t& input) const override {
-    vector_t dxdt(STATE_DIM);
+    vector_t dxdt(STATE_DIM(armInputDim_));
```
Comment (grizzi) [inline moma_controllers/moma_ocs2/include/moma_ocs2/MobileManipulatorPinocchioMapping.h:55]: Why is this initialized with dim `armInputDim_` ? The dimension does not account for the base

File: moma_controllers/moma_ocs2/src/MobileManipulatorTarget.cpp [hunk 2]
```diff
@@ -44,7 +44,7 @@ TargetTrajectories goalPoseToTargetTrajectories(const Eigen::Vector3d& position,
   const vector_t target = (vector_t(7) << position, orientation.coeffs()).finished();
   const vector_array_t stateTrajectory{target};
   // input trajectory
-  const vector_array_t inputTrajectory{vector_t::Zero(INPUT_DIM)};
+  const vector_array_t inputTrajectory{vector_t::Zero(INPUT_DIM(7 /* TODO: HARD-CODED TO 7 ROBOT ARM JOINTS */))};
```
Comment (grizzi) [inline moma_controllers/moma_ocs2/src/MobileManipulatorTarget.cpp:47]: We can actually get the right input dimension relying on the observation. In fact the observation should have an input field from whose length we can recover the input_dim

**Comment by grizzi** [comment]
Well done! I do not have any major comment. The only thing is that we have quite some code duplication just for the ur robot (which is also using a standard VelocityInterface).

**Comment by jk-ethz** [comment]
> Well done! I do not have any major comment. The only thing is that we have quite some code duplication just for the ur robot (which is also using a standard VelocityInterface).

Yes, there is a lot of overlap between `panda_mpc` and `ur_mpc`, which could be abstracted or commonized into `ocs2_mpc_ros`.
I think that's something one could address in a separate PR, since such a change would have made reviewing this PR more difficult.

**Comment by grizzi** [comment]
Yes, lets keep it like that for now. Maybe other changes between the two robots are needed which are not visible at the moment, given the limited testing.

**Pull request merged**